test/unittests/execute_floating_point_test.cpp

```diff
@@ -71,6 +71,54 @@ TEST(execute_floating_point, f64_add)
     EXPECT_THAT(execute(*instance, 0, {1.0011, 6.0066}), Result(7.0077));
 }
 
+TEST(execute_floating_point, f64_promote_f32)
+{
+    /* wat2wasm
+    (func (param f32) (result f64)
+      local.get 0
+      f64.promote_f32
+    )
+    */
+    auto wasm = from_hex("0061736d0100000001060160017d017c030201000a070105002000bb0b");
+    auto instance = instantiate(parse(wasm));
+
+    const std::pair<float, double> test_cases[] = {
+        {0.0f, 0.0},
+        {-0.0f, -0.0},
+        {1.0f, 1.0},
+        {-1.0f, -1.0},
+        {FP32::Limits::lowest(), double{FP32::Limits::lowest()}},
+        {FP32::Limits::max(), double{FP32::Limits::max()}},
+        {FP32::Limits::min(), double{FP32::Limits::min()}},
+        {FP32::Limits::denorm_min(), double{FP32::Limits::denorm_min()}},
+        {FP32::Limits::infinity(), FP64::Limits::infinity()},
+        {-FP32::Limits::infinity(), -FP64::Limits::infinity()},
+
+        // The canonical NaN must result in canonical NaN (only the top bit set).
+        {FP32::nan(FP32::canon), FP64::nan(FP64::canon)},
```

Comment:
Add `-FP32::nan(FP32::canon)` case

Reply:
And anan cases?

Reply:
> Add `-FP32::nan(FP32::canon)` case

Added.

> And anan cases?

Outside of the loop as they require sophisticated checks.